app/src/main/java/com/chesire/nekome/ActivityViewModel.kt

```diff
@@ -20,10 +24,24 @@ import kotlin.coroutines.CoroutineContext
 class ActivityViewModel @Inject constructor(
     private val repo: AccessTokenRepository,
     private val logoutHandler: LogoutHandler,
+    private val settings: ApplicationSettings,
     @IOContext private val ioContext: CoroutineContext,
     userRepository: UserRepository
 ) : ViewModel() {
 
+    private val navigationChannel = Channel<NavDirections>(Channel.BUFFERED)
+    val navigation = navigationChannel.receiveAsFlow()
+    private val snackBarChannel = Channel<Unit>()
+    val snackBar = snackBarChannel.receiveAsFlow()
```

Comment:
Would it be possible to do these with LiveData instead of flow? Just because other areas of the app still make use of LiveData from the view model. Maybe the LiveEvent class could replicate the "fire once" functionality?

Reply:
Oh thats nice. Just tried it and all works.